aten/src/ATen/cuda/CUDABlas.cpp

```diff
@@ -236,8 +236,287 @@ namespace at::cuda::blas {
     CUDABLAS_NONNEGINT_CHECK(bgemm<Dtype>, num_batches);  \
   } while (0)
 
+#if (!defined(USE_ROCM) && !defined(_MSC_VER)) || (defined(USE_ROCM) && ROCM_VERSION >= 50700)
```

Comment:
Why `!defined(USE_ROCM) && !defined(_MSC_VER))`?

Reply:
This check existed to say cublaslt cannot be used on Windows or ROCm builds.  In ROCm 5.7, hipblaslt support was added.  Perhaps it is more clear to write it as

`#if (defined(CUDA_VERSION) && !defined(_MSC_VER)) || (defined(USE_ROCM) && ROCM_VERSION >= 50700)`